crates/holochain/src/conductor/conductor.rs

```diff
@@ -3028,16 +3054,64 @@ mod misc_impls {
         /// Inject records into a source chain for a cell.
         /// If the records form a chain segment that can be "grafted" onto the existing chain, it will be.
         /// Otherwise, a new chain will be formed using the specified records.
+        //
+        // TODO: could integrate this better with [`SourceChain::flush`] when re-visit our workflows
```

Comment:
Can we work out now what's not ideal about this? It's not clear to me what "better" looks like or as a consequence, what is missing now to make this better